src/Reports/Full.php

```diff
@@ -115,36 +115,49 @@ public function generateFileReport($report, File $phpcsFile, $showSources=false,
         $maxErrorSpace = ($width - $paddingLength - 1);
 
+        $beforeMsg = '';
+        $afterMsg  = '';
+        if ($showSources === true) {
+            $beforeMsg = "\033[1m";
+            $afterMsg  = "\033[0m";
+        }
+
         foreach ($report['messages'] as $line => $lineErrors) {
             foreach ($lineErrors as $column => $colErrors) {
                 foreach ($colErrors as $error) {
-                    $message  = $error['message'];
-                    $msgLines = [$message];
-                    if (strpos($message, "\n") !== false) {
-                        $msgLines = explode("\n", $message);
-                    }
+                    $errorMsg = wordwrap(
+                        $error['message'],
+                        $maxErrorSpace
+                    );
 
-                    $errorMsg = '';
-                    $lastLine = (count($msgLines) - 1);
-                    foreach ($msgLines as $k => $msgLine) {
-                        if ($k === 0) {
-                            if ($showSources === true) {
-                                $errorMsg .= "\033[1m";
-                            }
-                        } else {
-                            $errorMsg .= PHP_EOL.$paddingLine2;
-                        }
+                    // Add the padding _after_ the wordwrap as the message itself may contain line breaks
+                    // and those lines will also need to receive padding.
+                    $errorMsg = str_replace("\n", $afterMsg.PHP_EOL.$paddingLine2.$beforeMsg, $errorMsg);
+                    $errorMsg = $beforeMsg.$errorMsg.$afterMsg;
 
-                        if ($k === $lastLine && $showSources === true) {
-                            $msgLine .= "\033[0m".' ('.$error['source'].')';
+                    if ($showSources === true) {
+                        $lastMsg          = $errorMsg;
+                        $startPosLastLine = strrpos($errorMsg, $paddingLine2.$beforeMsg);
+                        if ($startPosLastLine !== false) {
+                            // Message is multiline.
+                            $lastMsg = substr($errorMsg, ($startPosLastLine + strlen($paddingLine2.$beforeMsg)));
                         }
 
-                        $errorMsg .= wordwrap(
-                            $msgLine,
-                            $maxErrorSpace,
-                            PHP_EOL.$paddingLine2
-                        );
-                    }
+                        // When show sources is used, the message itself will be bolded, so we need to correct the length.
+                        $sourceSuffix = '('.$error['source'].')';
+
+                        $lastMsgPlusSourceLength = strlen($lastMsg);
+                        // Add space + source suffix length.
+                        $lastMsgPlusSourceLength += (1 + strlen($sourceSuffix));
+                        // Correct for the color codes.
+                        $lastMsgPlusSourceLength -= 8;
```

Comment:
I wonder if we should do like this.
```suggestion
                        $lastMsgPlusSourceLength -= strlen($beforeMsg.$afterMsg);
```
Disadvantage is that it has to measure the length of the string, advantage is that it won't accidentally break if someday someone changes the codes to something with a different length.

Or better might be to move the `strlen` outside of the loop (up near line 124).

Reply:
Agreed. Good change. Added now.